File: packages/cli/src/commands/logs.ts
import { getConfig } from '../utils/config.js';
import { getContainerStatus, streamLogs } from '../utils/docker.js';
import { streamNativeLogs } from '../utils/native.js';
import { log } from '../utils/ui.js';

export type LogsOptions = {
  follow?: boolean;
  tail?: string;
  native?: boolean;
};

export async function logsCommand(options: LogsOptions): Promise<void> {
  if (options.native) {
    await streamNativeLogs({ follow: Boolean(options.follow), tail: options.tail ?? '100' });
    return;
  }

  const { containerName } = getConfig();
  const status = await getContainerStatus(containerName);

  if (!status.exists) {
    log.error('Dillinger container does not exist.');
    process.exit(1);
  }

  await streamLogs(containerName, Boolean(options.follow), options.tail ?? '100');
}

File: packages/cli_native/src/commands/logs.rs
use anyhow::Result;

use crate::utils::{
    config::get_config,
    docker::{get_container_status, stream_logs},
    ui::LOG,
};

pub async fn logs_command(follow: bool, tail: &str) -> Result<()> {
    let config = get_config();
    let container_name = &config.container_name;
    let status = get_container_status(container_name).await?;

    if !status.exists {
        LOG.error("Dillinger container does not exist.");
        std::process::exit(1);
    }

    stream_logs(container_name, follow, tail).await
}
